library/core/src/slice/ascii.rs

```diff
@@ -27,6 +27,15 @@ impl [u8] {
         self.len() == other.len() && iter::zip(self, other).all(|(a, b)| a.eq_ignore_ascii_case(b))
     }
 
+    /// Checks if all bytes are ASCII decimal digit
+    #[inline]
+    pub fn is_ascii_digit(&self) -> bool {
+        self.iter().all(|i| {
```

Comment:
<https://doc.rust-lang.org/std/primitive.u8.html#method.is_ascii_digit> even takes `&u8`, which says to me that this can be written `x.iter().all(u8::is_ascii_digit)`, which seems perfectly fine, and not in need of an addition on `&[u8]`.